chaincode/algo_aggregate.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 Owkin, inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"chaincode/errors"
+)
+
+// Set is a method of the receiver Aggregate. It uses inputAggregateAlgo fields to set the AggregateAlgo
+// Returns the aggregateAlgoKey
+func (algo *AggregateAlgo) Set(db LedgerDB, inp inputAggregateAlgo) (algoKey string, err error) {
+	algoKey = inp.Hash
+	// find associated owner
+	owner, err := GetTxCreator(db.cc)
+	if err != nil {
+		return
+	}
+
+	permissions, err := NewPermissions(db, inp.Permissions)
+	if err != nil {
+		return
+	}
+
+	algo.AssetType = AggregateAlgoType
+	algo.Name = inp.Name
+	algo.StorageAddress = inp.StorageAddress
+	algo.Description = &HashDress{
+		Hash:           inp.DescriptionHash,
+		StorageAddress: inp.DescriptionStorageAddress,
+	}
+	algo.Owner = owner
+	algo.Permissions = permissions
+	return
+}
+
+// -------------------------------------------------------------------------------------------
+// Smart contracts related to an algo
+// -------------------------------------------------------------------------------------------
+// registerAggregateAlgo stores a new algo in the ledger.
+// If the key exists, it will override the value with the new one
```

Comment:
It's no longer the case for any type of algo. When we add the algo in the ledger [line 61](https://github.com/SubstraFoundation/substra-chaincode/blob/5dc099c59b2cb5c1a147f2b134f07f4fae7b21ce/chaincode/algo_aggregate.go#L66), we return directly if there is an error and there is one if the key already exists.
An issue should be opened to discuss if you should either remove this line from the `resgisterXXXAlgo` functions or if we should call `db.Put` rather than `db.Add`.